site/source/docs/compiling/Deploying-Pages.rst

```diff
@@ -0,0 +1,162 @@
+
+The following sections offer tips for improving the site experience. 
+
+Optimizing Download Sizes
+=========================
+
+The biggest slowdown to speedy page loading is most often the need to download large amounts of asset data for the project, especially if the page is heavy on WebGL textures or geometry. Compiled code generally takes up more space than handwritten JavaScript, but machine code compresses efficiently. Therefore when hosting asm.js and WebAssembly, it is critical to ensure that all content is transferred using gzip compression, which all browsers and CDNs nowadays support built-in. Gzip compressing ``.wasm`` files give on average 60-75% size reductions compared to uncompressed ones, so it practically never makes sense to serve uncompressed files.
+
+- To serve gzip-compressed assets on a CDN, use a gzip compression tool and precompress asset files offline before uploading to the CDN. Some web servers support compressing files on the fly, but for static asset content, that should be avoided since it can be costly for the server CPU to keep recompressing the files. Adjust the configuration of the web server to host the precompressed files with the HTTP response header ``Content-Encoding: gzip``. This instructs web browsers that the downloaded content should be transparently uncompressed before handing the data off to the page itself.
+
+- WebAssembly has now shipped in `Firefox 52 and Chrome 57 <http://caniuse.com/#feat=wasm>`_. Emscripten also supports targeting WebAssembly, by using the ``-s BINARYEN=1`` linker flag. WebAssembly is an evolution of asm.js, and if your project already successfully compiles to asm.js, it is likely to already work with WebAssembly as well. Compressed WebAssembly output files are on average 50-60% smaller than compressed asm.js files, so the benefit is large.
+
+- Make sure that gzip compression does not confuse the MIME types that the assets are served with. All JavaScript files (precompressed or not) should be best served with the HTTP response header ``Content-Type: application/javascript``, and all asset files (``.data``, ``.mem``) should be served with the header ``Content-Type: application/octet-stream``. WebAssembly ``.wasm`` files should be served with ``Content-Type: application/wasm``.
+
+- Try to minimize the amount of preloaded asset data that is downloaded up front with the Emscripten ``--preload-file`` linker flag. This data file is loaded up before the Emscripten compiled application starts to execute ``main()`` function at all, so all files that are stored in this package can greatly slow down the time to start up. Prefer to break up downloaded asset files to multiple separate packages and using asynchronous asset download APIs in Emscripten which can operate while the application is running.
```

Comment:
This advice is sort of at odds with the recommendation to asynchronously compile wasm and overlap that with asset download (are we making that recommendation here? if not, we should).